python/xgbserver/xgbserver/model.py

```diff
@@ -60,9 +60,12 @@ def load(self) -> bool:
     def predict(self, payload: Union[Dict, InferRequest], headers: Dict[str, str] = None) -> Union[Dict, InferResponse]:
         try:
             # Use of list as input is deprecated see https://github.com/dmlc/xgboost/pull/3970
+            predictions = []
             instances = get_predict_input(payload)
-            dmatrix = xgb.DMatrix(instances, nthread=self.nthread)
-            result = self._booster.predict(dmatrix)
-            return get_predict_response(payload, result, self.name)
+            for instance in instances:
+                dmatrix = xgb.DMatrix(instance, nthread=self.nthread)
+                result = self._booster.predict(dmatrix)
+                predictions.append(result)
```

Comment:
Why do we need to process individually ? in the old code `instances` and `result` itself are supposed to be an array.

Reply:
Passing in as batched instances would be more efficient than calling predict per instance